apps/expo-go/android/tools/src/main/java/host/exp/exponent/tools/ReactAndroidCodeTransformer.java

```diff
@@ -263,7 +262,7 @@ public Node visit(String methodName, MethodDeclaration n) {
       @Override
       String modifySource(String source) {
         // Make DevInternalSettings class "public"
-        source = source.replace("\nclass DevInternalSettings", "\npublic class DevInternalSettings");
+        source = source.replace("\ninternal class DevInternalSettings", "\npublic class DevInternalSettings");
```

Comment:
We'll also need to change the visibility of some of the methods in this class because React Native uses `explicitApi()` mode in kotlin.
- `onInternalSettingsChanged` needs to be public
- `addMenuItem` needs to add an explicit return type `Unit`
- The `Listener` interface and it's method both need `public` 
- The companion object needs `public`

Reply:
I did this but maybe there's a better way
```java
source = source.replace("companion object", "public companion object");
source = source.replace("interface Listener", "public interface Listener");
source = source.replace("fun onInternalSettingsChanged()", "public fun onInternalSettingsChanged()");
source = source.replace("override fun addMenuItem(title: String) = Unit", "override fun addMenuItem(title: String): Unit = Unit");
```
We'll need a something better for parsing and modifying kotlin

Reply:
Yeah I think for the sake of time we will need to go with the `replace` approach for now but for the next SDK release it would be interesting to add proper support for traversing kotlin methods

Reply:
updated